src/main/java/com/fasterxml/jackson/core/io/NumberInput.java

```diff
@@ -578,6 +578,7 @@ public static BigInteger parseBigIntegerWithRadix(final String s, final int radi
      * @since 2.17
      */
     public static boolean looksLikeValidNumber(final String s) {
-        return (s != null) && PATTERN_FLOAT.matcher(s).matches();
+        // the PATTERN_FLOAT match returns true for empty strings, so we need to check for digits separately
```

Comment:
No need: caller will fail on some things that look like number but aren't. Will remove added check.